### PR TITLE
Updates for TypeMap

### PR DESCRIPTION
**Pull request by ebolyen**

### CODE REVIEW

**Comment by ebolyen** [comment]
fixes #202 (removes option)
fixes #185 
fixes #71 (not as relevant as before anymore)
fixes #50 (feature removed)

**Comment by ebolyen** [comment]
fixes #210

**Comment by thermokarst** [comment]
Tests pass locally.

File: q2cli/click/command.py [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+
+            if args and not ctx.allow_extra_args and not ctx.resilient_parsing:
+                errors.append(click.UsageError(
+                    'Got unexpected extra argument%s (%s)'
+                    % (len(args) != 1 and 's' or '',
+                       ' '.join(map(click.core.make_str, args)))))
+        if errors:
+            click.echo(ctx.get_help()+"\n", err=True)
+            click.secho(
+                'There were some problem(s) with this command'.center(78, ' '),
```
Comment (thermokarst) [inline q2cli/click/command.py]: Do you want to deal with the inflection here? I can go either way.

**Pull request merged**